packages/commerce-sdk-react/src/components/StorefrontPreview/storefront-preview.tsx

```diff
@@ -23,14 +24,15 @@ type GetToken = () => string | undefined | Promise<string | undefined>
 export const StorefrontPreview = ({
     children,
     enabled = true,
-    getToken
+    getToken,
+    onContextChange
 }: React.PropsWithChildren<
-    // getToken is required unless enabled is false
-    {enabled?: true; getToken: GetToken} | {enabled: false; getToken?: GetToken}
+    // getToken and onContextChange are only required when enabled = true.
+    {enabled?: true; getToken: GetToken; onContextChange?: ContextChangeHandler;} | {enabled: false; getToken?: GetToken; onContextChange?: ContextChangeHandler;}
```

Comment:
```suggestion
    // Props are only required when Storefront Preview is enabled
    OptionalWhenDisabled<{getToken: GetToken; onContextChange?: ContextChangeHandler}>
```
Along with this helper type:
```ts
type OptionalWhenDisabled<T> = ({enabled?: true} & T) | ({enabled: false} & Partial<T>)
```

1. Make the comment evergreen
2. Use a helper type so the intention is clear and to avoid repetition